ur-packages/crypto-coin-identity/src/CoinIdentity.ts

```diff
@@ -73,12 +73,79 @@ export class CryptoCoinIdentity extends RegistryItem {
   public getType = () => this.type
   public getSubType = () => this.subtype
 
+  public getParent = () => {
+    // If we dont have any subtypes, return null
+    if (!this.subtype.length) return null;
+
+    // Otherwise remove latests subtype and return a new CryptoCoinIdentity
+    const subtypes = this.subtype.slice(1, this.subtype.length)
+    return new CryptoCoinIdentity(this.curve, this.type, subtypes)
+  }
+
+
+  // Make an Iterator that returns all the parents of this CryptoCoinIdentity
+  getAllParents2(): Iterable<CryptoCoinIdentity> {
+    let currentParent = this.getParent();
+
+    const parentIterator = {
+      [Symbol.iterator](): Iterator<CryptoCoinIdentity> {
+        return {
+          next(): IteratorResult<CryptoCoinIdentity> {
+            if ( currentParent ) {
+              const returnParent = new CryptoCoinIdentity(currentParent.getCurve(), currentParent.getType(), currentParent.getSubType());
+              currentParent = currentParent.getParent();
+
+              return {
+                value: returnParent,
+                done: false,
+              }
+            }
+            return { value: undefined, done: true }
+          },
+        }
+      }
+    };
+
+    return parentIterator;
+  }
```

Comment:
this can be removed?

Reply:
Check the tests and time it tooks, I think it is better to remove `getAllParents` instead of this one